ns for fx: add option to skip matching classes and functions

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #56927 ns for fx: add missing add_relu and mul_relu patterns
* #56762 ns for fx: additional bugfix for user defined functions
* #56742 ns for fx: allow comparing int8 to int8 for functionals
* **#56493 ns for fx: add option to skip matching classes and functions**
* #56408 ns for fx: support binary ops when adding unshadowed loggers for inputs
* #56384 ns for fx: bug fix for shadowing fp16 emulation patterns
* #56311 ns for fx: add fp16 function shadowing
* #56301 ns for fx: allow user functions in shadowing
* #56296 ns for fx: move node I/O dtype mapping to be local instead of global
* #56292 ns for fx: handling for user functions in weight and unshadowed act APIs
* #56283 ns for fx: enable user defined functions for graph matching

Summary:

Adds a config option to skip matching classes by class type
and functions by function type.

This is useful when users make custom modules which return
types other than tensors. With the current implementation of
Logger, these are not scriptable.

Test Plan:

```
python test/test_quantization.py TestFXNumericSuiteCoreAPIs.test_user_module_scriptable
```

needs more testing before land

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27886107](https://our.internmc.facebook.com/intern/diff/D27886107)

vkuzo committed:
ns for fx: add option to skip matching classes and functions

Summary:

Adds a config option to skip matching classes by class type
and functions by function type.

This is useful when users make custom modules which return
types other than tensors. With the current implementation of
Logger, these are not scriptable.

Test Plan:

```
python test/test_quantization.py TestFXNumericSuiteCoreAPIs.test_user_module_scriptable
```

needs more testing before land

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo committed:
Update on "ns for fx: add option to skip matching classes and functions"


Summary:

Adds a config option to skip matching classes by class type
and functions by function type.

This is useful when users make custom modules which return
types other than tensors. With the current implementation of
Logger, these are not scriptable.

Test Plan:

```
python test/test_quantization.py TestFXNumericSuiteCoreAPIs.test_user_module_scriptable
```

needs more testing before land

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27886107](https://our.internmc.facebook.com/intern/diff/D27886107)

[ghstack-poisoned]

jerryzh168 reviewed (approved):
please fix CI before land

vkuzo committed:
Update on "ns for fx: add option to skip matching classes and functions"


Summary:

Adds a config option to skip matching classes by class type
and functions by function type.

This is useful when users make custom modules which return
types other than tensors. With the current implementation of
Logger, these are not scriptable.

Test Plan:

```
python test/test_quantization.py TestFXNumericSuiteCoreAPIs.test_user_module_scriptable
```

needs more testing before land

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27886107](https://our.internmc.facebook.com/intern/diff/D27886107)

[ghstack-poisoned]

vkuzo committed:
Update on "ns for fx: add option to skip matching classes and functions"


Summary:

Adds a config option to skip matching classes by class type
and functions by function type.

This is useful when users make custom modules which return
types other than tensors. With the current implementation of
Logger, these are not scriptable.

Test Plan:

```
python test/test_quantization.py TestFXNumericSuiteCoreAPIs.test_user_module_scriptable
```

needs more testing before land

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27886107](https://our.internmc.facebook.com/intern/diff/D27886107)

[ghstack-poisoned]

vkuzo committed:
Update on "ns for fx: add option to skip matching classes and functions"


Summary:

Adds a config option to skip matching classes by class type
and functions by function type.

This is useful when users make custom modules which return
types other than tensors. With the current implementation of
Logger, these are not scriptable.

Test Plan:

```
python test/test_quantization.py TestFXNumericSuiteCoreAPIs.test_user_module_scriptable
```

needs more testing before land

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27886107](https://our.internmc.facebook.com/intern/diff/D27886107)

[ghstack-poisoned]

vkuzo committed:
Update on "ns for fx: add option to skip matching classes and functions"


Summary:

Adds a config option to skip matching classes by class type
and functions by function type.

This is useful when users make custom modules which return
types other than tensors. With the current implementation of
Logger, these are not scriptable.

Test Plan:

```
python test/test_quantization.py TestFXNumericSuiteCoreAPIs.test_user_module_scriptable
```

needs more testing before land

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27886107](https://our.internmc.facebook.com/intern/diff/D27886107)

[ghstack-poisoned]
